base/src/main/java/io/spine/validate/FieldValidator.java

```diff
@@ -138,6 +143,18 @@ boolean fieldValueNotSet() {
         return result;
     }
 
+    private void checkCanBeRequired() {
+        boolean fieldIsRequired = isRequiredField();
+        if (!canBeRequired && fieldIsRequired) {
+            String messageFormat = "Fields of type %s can't be declared as `(required)`.";
+            String typeName = this.field()
+                                  .descriptor()
+                                  .getType()
+                                  .name();
```

Comment:
This is ugly. Please use `FieldDeclaration` methods. If there is no such method, please add one.